cmake/gen_version_h.cmake

```diff
@@ -2,33 +2,23 @@
 
 cmake_minimum_required(VERSION 3.20.0)
 
+include(${ZEPHYR_BASE}/cmake/git.cmake)
+
 if(VERSION_TYPE STREQUAL KERNEL)
   set(BUILD_VERSION_NAME BUILD_VERSION)
 else()
   set(BUILD_VERSION_NAME ${VERSION_TYPE}_BUILD_VERSION)
 endif()
 
+cmake_path(GET VERSION_FILE PARENT_PATH work_dir)
+
 if(NOT DEFINED ${BUILD_VERSION_NAME})
-  cmake_path(GET VERSION_FILE PARENT_PATH work_dir)
-  find_package(Git QUIET)
-  if(GIT_FOUND)
-    execute_process(
-      COMMAND ${GIT_EXECUTABLE} describe --abbrev=12 --always
-      WORKING_DIRECTORY                ${work_dir}
-      OUTPUT_VARIABLE                  ${BUILD_VERSION_NAME}
-      OUTPUT_STRIP_TRAILING_WHITESPACE
-      ERROR_STRIP_TRAILING_WHITESPACE
-      ERROR_VARIABLE                   stderr
-      RESULT_VARIABLE                  return_code
-    )
-    if(return_code)
-      message(STATUS "git describe failed: ${stderr}")
-    elseif(NOT "${stderr}" STREQUAL "")
-      message(STATUS "git describe warned: ${stderr}")
-    endif()
-  endif()
+  git_describe(${work_dir} ${BUILD_VERSION_NAME})
 endif()
 
+git_commit_hash(${work_dir} GIT_COMMIT_HASH)
+git_commit_hash_short(${GIT_COMMIT_HASH} GIT_COMMIT_HASH_SHORT GIT_COMMIT_HASH_SHORT_INT)
```

Comment:
these do not take into consideration the prefix, as above

Reply:
All variables defined in this file are temporary, they don't exist in 5 lines time.
The only requirement is that the name matches that in `version.h.in`.
`version.h.in` does add the prefix in the generated `version.h` file.
I can certainly add the prefix, at the cost of making the code much less clear through the additional dereferencing.
I will note that even the code above only adds the prefix "some" of the time `VERSION_TYPE != KERNEL`.

Reply:
The reason they exist for the other variables is that they are generated by `version.cmake`, which IS included from the main build system and `KERNEL` and `APP` need to co-exist. Not the case here.

Reply:
UPPERCASE VARIABLES = global
lowercase variables = temp

Reply:
My wording was imprecise.
They are temporary because the CMake invocation terminates in 5 lines time, not temporary in the sense they are not used anywhere else.
As I mentioned, they are used in `version.h.in` as part of the `string(CONFIGURE "${version_content}" version_content)` line.
I can add whatever prefixes are desired, I just see little value because of my first point.